test/document_stores/test_weaviate.py

```diff
@@ -105,3 +109,14 @@ def test_query(document_store_with_docs):
 
     docs = document_store_with_docs.query(filters={"content": ["live"]})
     assert len(docs) == 3
+
+
+@pytest.mark.weaviate
+@pytest.mark.parametrize("document_store_with_docs", ["weaviate"], indirect=True)
+def test_get_all_documents(document_store_with_docs):
+    # Ensure the method works no matter the value of QUERY_MAXIMUM_RESULTS
+    # see https://github.com/deepset-ai/haystack/issues/2517
+    with patch.object(document_store_with_docs, "get_document_count") as count:
+        count.return_value = 13_000
+        docs = document_store_with_docs.get_all_documents()
+        assert len(docs) == 3
```

Comment:
A couple of minor things:
 
- The test name is not very descriptive of that the test is testing. I love endless test names like `test_get_all_documents_unaffected_by_querymaximumresults` :grin: Makes obvious that this is not just testing `get_all_documents`, and not testing it as a whole, but just a corner case. It also highlights that this is not an integration test.
  - I see that you tried to fit with the style of the existing tests. However, our tests in general don't look they way I would like them to :sweat_smile: By just looking that the rest of them I can see many other things that should be changed (like, why to we have a parametrized fixture `document_store_with_docs`, with just one value, `weaviate`?? etc). I think it's no problem if new tests take a complete different direction than the old ones. Actually the earlier they do, the better.

- We generally favor pytest-style tests, so a `monkeypatch` with a lambda, or something similar, rather than `patch`. Very minor tho, you can leave it if you like it more. In the future I think we should choose which one to use.

Reply:
I've removed the markers altogether, I'll fix the rest of the test functions in a different PR

Reply:
Watch out, the `weaviate` marker technically is useful. Either the test is marked explicitly, or you should add the string `weaviate` in the test name of the test uses Weaviate. It allows users to filter for Weaviate tests (or filter them out) without having to exclude files by name

Reply:
Yeah I wanted to mark the whole module instead of single functions, let me add it back for now, will fix it later.